cider/function/FunctionLookup.cpp

```diff
@@ -0,0 +1,189 @@
+  auto iter = scalar_function_look_up_ptr_map_.find(from_platform);
+  if (iter == scalar_function_look_up_ptr_map_.end()) {
+    return nullptr;
+  }
+  SubstraitFunctionLookupPtr scalar_function_look_up_ptr = iter->second;
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      scalar_function_look_up_ptr->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return function_mappings_->getFunctionOpSupportType(func_name);
+  }
+  return nullptr;
+}
+
+const OpSupportExprTypePtr FunctionLookup::getAggFunctionOpSupportType(
+    const FunctionSignature& function_signature) const {
+  const std::string& from_platform = function_signature.from_platform;
+  auto iter = aggregate_function_look_up_ptr_map_.find(from_platform);
+  if (iter == aggregate_function_look_up_ptr_map_.end()) {
+    return nullptr;
+  }
+  SubstraitFunctionLookupPtr agg_function_look_up_ptr = iter->second;
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      agg_function_look_up_ptr->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return function_mappings_->getFunctionOpSupportType(func_name);
+  }
+  return nullptr;
+}
+
+const OpSupportExprTypePtr FunctionLookup::getExtensionFunctionOpSupportType(
+    const FunctionSignature& function_signature) const {
+  const std::string& from_platform = function_signature.from_platform;
+  auto iter = extension_function_look_up_ptr_map_.find(from_platform);
+  if (iter == extension_function_look_up_ptr_map_.end()) {
+    return nullptr;
+  }
+  SubstraitFunctionLookupPtr extension_function_look_up_ptr = iter->second;
+  const std::string& func_name = function_signature.func_name;
+  const auto& functionSignature =
+      cider::function::substrait::SubstraitFunctionSignature::of(
+          func_name, function_signature.arguments, function_signature.return_type);
+  const auto& functionOption =
+      extension_function_look_up_ptr->lookupFunction(functionSignature);
+  if (functionOption.has_value()) {
+    return std::make_shared<OpSupportExprType>(OpSupportExprType::FunctionOper);
+  }
+  return nullptr;
+}
+
+const OpSupportExprTypePtr FunctionLookup::getFunctionOpSupportType(
+    const FunctionSignature& function_signature) const {
+  OpSupportExprTypePtr result_ptr = nullptr;
+  result_ptr = getScalarFunctionOpSupportType(function_signature);
+  if (result_ptr) {
+    return result_ptr;
+  }
+  result_ptr = getAggFunctionOpSupportType(function_signature);
```

Comment:
The rule is that we should check extension functions first.

Reply:
Done.